pkg/clients/handlers.go

```diff
@@ -0,0 +1,166 @@
+package clients
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/responses"
+	"github.com/go-chi/chi/v5"
+	hClient "github.com/ory/hydra-client-go/v2"
+)
+
+type API struct {
+	service ServiceInterface
+
+	logger logging.LoggerInterface
+}
+
+func (a *API) RegisterEndpoints(mux *chi.Mux) {
+	mux.Get("/clients", a.ListClients)
+	mux.Post("/clients", a.CreateClient)
+	mux.Get("/clients/{client_id}", a.GetClient)
+	mux.Put("/clients/{client_id}", a.UpdateClient)
+}
+
+func (a *API) WriteJSONResponse(w http.ResponseWriter, data interface{}, msg string, status int, links interface{}, meta interface{}) {
+	w.Header().Set("Content-Type", "application/json")
+	r := new(responses.Response)
+
+	r.Data = data
+	r.Message = msg
+	r.Status = status
+	r.Links = links
+	r.Meta = meta
+
+	rr, err := json.Marshal(r)
+	if err != nil {
+		a.logger.Errorf("Unexpected error: %s", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.Write(rr)
+	w.WriteHeader(r.Status)
+}
```

Comment:
still not convinced about this, but right now cannot find a better - alternative without poking around further